src/lambda/local/debugConfigurationProvider.ts

```diff
@@ -0,0 +1,265 @@
+
+        let tasks: TasksConfig | undefined
+        if (await fileExists(tasksPath)) {
+            tasks = JSON.parse(await readFileAsString(tasksPath, 'utf8')) as TasksConfig | undefined
+        }
+
+        if (!tasks) {
+            tasks = {
+                version: '2.0.0'
+            }
+        }
+
+        if (!tasks.tasks) {
+            tasks.tasks = []
+        }
+
+        if (!tasks.tasks.some(t => t.label === label)) {
+            tasks.tasks.push(this.createPreLaunchTask(functionName, event, debugPort))
+        }
+
+        if (!await fileExists(configRoot) || !(await statAsync(configRoot)).isDirectory()) {
+            await mkdirAsync(configRoot)
+        }
+        // TODO: Read user's tab-width setting and use that instead of hard-coding '4'.
+        await writeFileAsync(tasksPath, JSON.stringify(tasks, undefined, 4))
+
+        return label
+    }
+
+    private createPreLaunchTask(
+        functionName: string,
+        event: { [ key: string ]: string } = {},
+        debugPort: number = 5858
+    ) {
+        return {
+            type: 'shell',
+            label: this.getTaskLabel(functionName),
+            command: 'echo',
+            args: [
+                `${this.escapeForBash(JSON.stringify(event))}`,
+                '|',
+                'sam',
+                'local',
+                'invoke',
+                `${this.escapeForBash(functionName)}`,
+                '-d',
+                `${debugPort}`
+            ],
+            windows: {
+                args: [
+                    `${this.escapeForPowerShell(JSON.stringify(event))}`,
+                    '|',
+                    'sam',
+                    'local',
+                    'invoke',
+                    `${this.escapeForPowerShell(functionName)}`,
+                    '-d',
+                    `${debugPort}`
+                ],
+            },
+            isBackground: true,
+            presentation: {
+                echo: true,
+                reveal: 'always',
+                focus: false,
+                panel: 'dedicated',
+                showReuseMessage: true
+            },
+            problemMatcher: {
+                owner: 'lambda-node',
+                // tslint:disable-next-line:no-invalid-template-strings
+                fileLocation: [ 'relative', '${workspaceFolder}' ],
+                pattern: [
+                    {
+                        // TODO: For now, use regex that never matches anything.
+                        // Update as we determine what issues we can recognize.
+                        regexp: '^(x)(\b)(x)$',
+                        file: 1,
+                        location: 2,
+                        message: 3
+                    }
+                ],
+                background: {
+                    activeOnStart: true,
+                    beginsPattern: String.raw`^Fetching lambci\/lambda:nodejs\d+\.\d+ Docker container image......$`,
```

Comment:
For these patterns, find out what the messages look like in other languages, and generalize regexes so that they aren't language-specific.

Reply:
SAM CLI is not currently localized.